### PR TITLE
Feature/add extra fields

### PR DESCRIPTION
**Pull request by BFoucher**
Add 2 processors:
- Domain processor for include extra fields in parameters and request uri

- Git processor for extract commit number and add in logs

### CODE REVIEW

File: DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -22,8 +22,14 @@ public function getConfigTreeBuilder()
 
         $rootNode
             ->children()
+                ->arrayNode('extra_fields')
```
Comment (lenybernard) [inline DependencyInjection/Configuration.php:25]: I think `context_extra_fields` is a better name to understand its role

File: Monolog/Processor/GitProcessor.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    private function getRevision()
+    {
+        if (self::$cache !== false) {
+            return self::$cache;
+        }
+
+        //Get revision number from .git
+        if ($gitRev = shell_exec('git rev-parse HEAD')) {
```
Comment (lenybernard) [inline Monolog/Processor/GitProcessor.php]: I'd surround the `git rev-parse HEAD` to be sure there is no problem running this command and its output is well a hash as expected

File: Monolog/Processor/GitProcessor.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Troopers\MetricsBundle\Monolog\Processor;
+
+class GitProcessor
```
Comment (lenybernard) [inline Monolog/Processor/GitProcessor.php:5]: a phpunit would be useful to be sure to test the `getRevision()` method

File: DependencyInjection/Configuration.php (hunk 1 above)
Comment (BFoucher) [reply DependencyInjection/Configuration.php:25]: this fields are added in 'extra' not in 'context', if you prefer i can rename it "extra" ?

File: Monolog/Processor/GitProcessor.php (hunk 3 above)
Comment (BFoucher) [reply Monolog/Processor/GitProcessor.php:5]: see f5c3f9f

**Review by lenybernard — APPROVED** [review]
👍 ✅ ❤️

**Pull request merged**